### ISSUE TITLE
dotnet cli not displaying opening square bracket plus two characters on Docker

### ISSUE THREAD

**Issue by DanSeefeldt** [dotnet/cli#2042]
## Steps to reproduce
1. Create a Docker container using a preview container
2. type `dotnet --help`

## Expected  behavior
Help should be fully displayed.

Also, the same issue occurs if I write an app.

`Console.WriteLine("[Hello]");` 

results in:

	ello]

## Actual behavior
Any text that contains square brackets is missing the leading square bracket and the following two letters.

Example:

    .NET Command Line Tools (1.0.0-beta-001793)
    Usage: dotnet mmon-options] mmand] guments]
    
    Arguments:
      mmand] The command to execute
      guments]   Arguments to pass to the command
    
    Common Options (passed before the command):
      -v|--verbose  Enable verbose output
      --version Display .NET CLI Version Info

## Environment data
`dotnet --version` output:
    
    .NET Command Line Tools (1.0.0-beta-001793)
    
    Product Information:
     Version: 1.0.0-beta-001793
     Commit Sha:  42a0eec967
    
    Runtime Environment:
     OS Name: ubuntu
     OS Version:  14.04
     OS Platform: Linux
     Runtime Id:  ubuntu.14.04-x64

**Comment by MichaelSimons** [comment]
I verified this repros with the latest bits (e.g. 1.0.0-beta-001988)

**Comment by brthor** [comment]
I have seen this frequently as well. Haven't had the time to investigate yet

In docker it can be pretty bad

**Comment by vlesierse** [comment]
I see the same on Mac OSX. Also my own .NET Core apps using the `Microsoft.Extensions.CommandLineUtils` shows this problem.

```bash
macbook:dotnet lesi290824$ dotnet
Usage: dotnet [--help | app.dll]
macbook:dotnet lesi290824$ dotnet --help
.NET Command Line Tools (1.0.0-beta-002058)
Usage: dotnet mmon-options] mmand] guments]

Arguments:
  mmand]     The command to execute
  guments]   Arguments to pass to the command
```

**Comment by MichaelSimons** [comment]
A couple observations I have made.

The following c# code will also show the truncation behavior.
```
[DllImport("libc")]
private static extern int printf(string format);
...
printf("[hello]\n");
```

If you do a similar thing in a C program, everything works as expected.

The default `LANG` within a docker container is `POSIX`.  If you change this to `en_US.UTF-8`, the issue goes away.  You can reproduce this truncation behavior outside of docker by changing the `LANG` to `POSIX`

**Comment by vlesierse** [comment]
I can confirm that `export LANG=en_US.UTF-8` in my Mac terminal will fix this behaviour.

**Comment by MichaelSimons** [comment]
I only see this misbehavior when using `dotnet run` on my app.  If I `dotnet build` and `dotnet app.dll`, the braces are not truncated.

**Comment by stephentoub** [comment]
I took a look at this.  

@ellismg, this is related to the coreclr implementation of IndexOf on strings.  When LANG=, ```"[hello]".IndexOf("\x1b[")``` returns 0 rather than -1, which it returns when LANG=en_US.UTF-8.  This then causes problems with https://github.com/dotnet/cli/blob/30b5d7a95298250e68f3884ee5c8361f3c021be7/src/Microsoft.DotNet.Cli.Utils/AnsiConsole.cs#L69, which will skip two characters past 0 rather than taking the path where the special sequence isn't found.

@piotrpMSFT, @brthor, until the coreclr issue is addressed, cli should be able to work around this by passing StringComparison.Ordinal as the second argument to IndexOf.

**Comment by xied75** [comment]
:+1: Rather annoying.

**Comment by kdaveid** [comment]
Issue fixed with 1.0.0-preview1-002702. LANG=de_CH.UTF-8 which is not en_US.UTF8.
Can anyone confirm? 

Issue could be closed

**Comment by ellismg** [comment]
Basically and non C or Empty locale should work.

We have made changes to the CLR to do the right thing even when this is empty, but they are not present in RC2 (they are in the nightly builds now and will be part of RTM).

I think this is safe to close.
